test/common/buffer/watermark_buffer_test.cc

```diff
@@ -433,17 +433,19 @@ TEST_F(WatermarkBufferTest, OverflowWatermarkDisabledOnVeryHighValue) {
   Runtime::LoaderSingleton::getExisting()->mergeValues({{"envoy.buffer.overflow_multiplier", "3"}});
   buffer1.setWatermarks((std::numeric_limits<uint32_t>::max() / 3) + 4);
 
-  // Add 2 halves instead of full uint32_t::max to get around std::bad_alloc exception
-  const uint32_t half_max = std::numeric_limits<uint32_t>::max() / 2;
-  const std::string half_max_str = std::string(half_max, 'a');
-  buffer1.add(half_max_str.data(), half_max);
-  buffer1.add(half_max_str.data(), half_max);
+  // Add many segments instead of full uint32_t::max to get around std::bad_alloc exception
+  const uint32_t segment_denominator = 32;
+  const uint32_t big_segment_len = std::numeric_limits<uint32_t>::max() / segment_denominator;
+  const std::string big_segment_str = std::string(big_segment_len, 'a');
+  for (uint32_t i = 0; i < segment_denominator; ++i) {
+    buffer1.add(big_segment_str.data(), big_segment_len);
```

Comment:
It may be possible to improve this test a bit by use of reserve() and commit() on the buffer to avoid actually allocating a huge strings and copying 4GB worth of data.  I guess leaving the copy in is probably fine, unless we have some concerns about runtime.